ui/packages/audio/src/Audio.svelte

```diff
@@ -29,13 +31,33 @@
 	// export let type: "normal" | "numpy" = "normal";
 
 	let recording = false;
-	let recorder: MediaRecorder;
+	let recorder: IMediaRecorder;
 	let mode = "";
-	let audio_chunks: Array<Blob> = [];
-	let audio_blob;
+	let header: Uint8Array | undefined = undefined;
+	let pending_stream: Array<Uint8Array> = [];
 	let player;
 	let inited = false;
 	let crop_values = [0, 100];
+	const STREAM_TIMESLICE = 500;
+	const NUM_HEADER_BYTES = 44;
+	let audio_chunks: Array<Blob> = [];
+	let audio_blob;
+	let module_promises:
+		| [
+				Promise<typeof import("extendable-media-recorder")>,
+				Promise<typeof import("extendable-media-recorder-wav-encoder")>
+		  ];
```

Comment:
We needed the types of the things we are actually importing, we can do this by doing `typeof <thing we want the type of>` in the case of `typeof import(...)` it gives us the type of the dynamically imported module (not of the return value of `import(module)`). Since `import()` always returns a promise we pass this to the `Promise<T>` generic, so TS knows it is a promise containing that module.